ras_backstage/resources/sign_in/sign_in.py

```diff
@@ -53,12 +53,28 @@ def post():
         username = message_json.get('username')
         password = message_json.get('password')
 
-        # Obviously horrible, stopgap until uaa is implemented
-        if username == current_app.config['USERNAME'] and password == current_app.config['PASSWORD']:
-            # We're assuming that uaa will return an Oauth2 token though it's almost certain that
-            # this will change once we know exactly what is being returned.
-            logger.info("Authentication successful", user=username)
-            return make_response(jsonify({"token": "1234abc"}), 201)
+        if not current_app.config.get('USE_UAA'):
+            logger.info('Retrieving sign-in details')
+            if username == current_app.config['USERNAME'] and password == current_app.config['PASSWORD']:
+                logger.info("Authentication successful", user=username)
+                return make_response(jsonify({"token": "1234abc"}), 201)
+            else:
+                logger.info("Authentication failed", user=username)
+                return make_response(jsonify({"error": "Username and/or password incorrect"}), 401)
         else:
-            logger.info("Authentication failed", user=username)
-            return make_response(jsonify({"error": "Username and/or password incorrect"}), 401)
+            logger.info('Retrieving sign-in details using UAA')
+            message_json = request.get_json()
+            username = message_json.get('username')
+            password = message_json.get('password')
+
+            oauth2_token = uaa_controller.sign_in(username, password)
+            token_expiry = datetime.now() + timedelta(seconds=int(oauth2_token['expires_in']))
+            oauth2_token['expires_at'] = token_expiry.timestamp()
```

Comment:
Can we set an expiry time in the uaa service or do we have to set it here?

Reply:
None of this is actually needed, we're going to use the token directly from the UAA